.circleci/config.yml

```
@@ -217,121 +173,133 @@ commands:
 
-  win_build:
-    description: Run build on Windows
+
+jobs:
+  e2e_tests:
     parameters:
       python_version:
         type: string
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
+    steps:
+      - setup_python_env
+      - install_node_dependencies
+      - run_e2e_tests
+  
+  win_e2e_tests:
+    parameters:
+      python_version:
+        type: string
+    executor: win/default
     steps:
       - checkout
-      # - restore_cache:
-      #     keys:
-      #       - v${CACHE_VERSION}-{{ arch }}-dependencies-{{ checksum "package.json" }}-{{ checksum "package/requirements.txt" }}-{{ checksum "package/test_requirements.txt" }}
-      #       # fallback to using the latest cache if no exact match is found
-      #       - v${CACHE_VERSION}-{{ arch }}-dependencies-
       - win_setup_conda:
           python_version: <<parameters.python_version>>
       - win_setup_requirements
       - install_node_dependencies
-      # - save_cache:
-      #     paths:
-      #       - node_modules
-      #       - c:\tools\miniconda3\envs\kedro-viz\
-      #       - c:\users\circleci\appdata\local\pip\cache\
-      #     key: v${CACHE_VERSION}-{{ arch }}-dependencies-{{ checksum "package.json" }}-{{ checksum "package/requirements.txt" }}-{{ checksum "package/test_requirements.txt" }}
-      - run:
-          name: Run Python unit tests
-          command: |
-            conda activate kedro-viz
-            rm -r build, package/dist, package/kedro_viz/html, pip-wheel-metadata, package/kedro_viz.egg-info -ErrorAction Ignore
-            npm run build
-            cp -R build package/kedro_viz/html
-            pytest package/tests --no-cov
-      - run:
-          name: Run e2e tests
-          command: |
-            conda activate kedro-viz
-            cd package
-            behave
-
-jobs:
-  build_37:
-    executor: circleci_machine_37
-    steps: [build]
+      - run_win_e2e_tests
 
-  build_38:
-    executor: circleci_machine_38
-    steps: [build]
-
-  build_39:
-    executor: circleci_machine_39
-    steps: [build]
-
-  build_310:
-    executor: circleci_machine_310
-    steps: [build]
-
-  deploy_demo:
-    executor: demo_executor
-    steps: [deploy_demo]
+  unit_tests:
+    parameters:
+      python_version:
+        type: string
+    executor:
+      name: docker
+      python_version: <<parameters.python_version>>
+    steps:
+      - checkout 
+      - setup_python_env
+      - install_node_dependencies
+      - run_python_tests
 
-  # Windows-related jobs
-  win_build_37:
-    executor: win/default
+  win_unit_tests:
     parameters:
       python_version:
         type: string
-    working_directory: ~/repo
+    executor: win/default
     steps:
-      - checkout
-      - win_build:
+      - checkout 
+      - win_setup_conda:
           python_version: <<parameters.python_version>>
+      - win_setup_requirements
+      - install_node_dependencies
+      - run_win_python_tests
```

Similarly we can just write this one inline, delete the definition of `run_win_python_tests` and reduce the number of redirections in the file.
```suggestion
      - run:
          name: Run Python tests on Windows
          command:
            conda activate kedro-viz; make pytest
```